app/imports/components/projects/AddProject.jsx

```diff
@@ -224,23 +213,48 @@ export class AddProject extends Component {
     }
 
     if (isValid) {
-      let customerIndex = _.findIndex(this.props.customers, { name: project.customer });
-
-      if (customerIndex === -1) {
-        let customer = { name: project.customer };
-
-        let customerId = addCustomer.call(customer, (err, res) => {
-          if (err) {
-            console.error(err.error);
-          } else {
-            project.customerId = customerId;
-            this.callAddProject(project);
-          }
-        });
-      } else {
-        project.customerId = this.props.customers[customerIndex]._id;
-        this.callAddProject(project);
-      }
+      addProject.call(project, (err, res) => {
+        if (err) {
+          console.error(err.error);
+        } else {
+          this.cleanForm();
+        }
+      });
+    }
+  }
+
+  onSubmitHandle(event) {
+    event.preventDefault();
+
+    let customerIndex = _.findIndex(this.props.customers, { name: this.state.customer });
+
+    if (customerIndex === -1) {
+      // Customer does not exist in props array of customers
+      let customer = { name: this.state.customer };
+
+      addCustomer.call(customer, (err, res) => {
+        if (err) {
+          console.error(err.error);
+        } else {
+          this.setState({
+            // set customer ID from method return value
+            customerId: res,
+          }, () => {
+            // after state set, validate project
+            this.validateProject();
```

Comment:
My idea was to move the part that adds the Customer into a server only function that gets called inside the method. I was thinking theres no need to run this process of adding a new customer in the front end in this case. 

Not sure about this, I need to check the code to be sure of what i'm proposing.